Python/pywarpx/picmi.py

```diff
@@ -445,26 +445,43 @@ def initialize_inputs(self):
 
 class ElectromagneticSolver(picmistandard.PICMI_ElectromagneticSolver):
     def init(self, kw):
-        assert self.method is None or self.method in ['Yee', 'CKC'], Exception("Only 'Yee' and 'CKC' FDTD are supported")
+        assert self.method is None or self.method in ['Yee', 'CKC', 'PSATD'], Exception("Only 'Yee', 'CKC' and 'PSATD' are supported")
 
         self.do_pml = kw.pop('warpx_do_pml', None)
         self.pml_ncell = kw.pop('warpx_pml_ncell', None)
+        self.v_galilean = kw.pop('warpx_v_galilean', None)
 
     def initialize_inputs(self):
 
         self.grid.initialize_inputs()
 
         pywarpx.warpx.do_pml = self.do_pml
         pywarpx.warpx.pml_ncell = self.pml_ncell
+        
+        pywarpx.psatd.v_galilean = self.v_galilean
 
         # --- Same method names are used, though mapped to lower case.
-        pywarpx.algo.maxwell_fdtd_solver = self.method
+        if self.method is not 'PSATD':
+            pywarpx.algo.maxwell_fdtd_solver = self.method
 
         if self.cfl is not None:
             pywarpx.warpx.cfl = self.cfl
 
         if self.source_smoother is not None:
             self.source_smoother.initialize_inputs(self)
+            
+        if self.l_nodal is not None:
+            pywarpx.warpx.do_nodal = self.l_nodal
+            
```

Comment:
There are also set here.
```suggestion
        if self.method == 'PSATD':
            pwwarpx.psatd.periodic_single_box_fft = self.periodic_single_box_fft
            pwwarpx.psatd.fftw_plan_measure = self.fftw_plan_measure
            pwwarpx.psatd.do_current_correction = self.do_current_correction
```